### PR TITLE
Add tests for `DeleteBucket`

### LINKED ISSUE #362: Test `DeleteBucket` Method in BucketRuntime Service
# Summary
Test `DeleteBucket` Method in BucketRuntime Service

## Description
Create test cases to validate the `DeleteBucket` method in the `BucketRuntime` service. Verify that it correctly deletes the specified bucket and handles cases where the bucket does not exist or cannot be deleted due to dependencies. Test error scenarios and check if the method provides meaningful error messages.

### Instructions:
1. Install the necessary CRDs in the test environment.
2. Write test cases to confirm the functionality of the `DeleteBucket` method.
3. Ensure that the tests only patch the `rook` resources for testing purposes.
4. Test scenarios should include successful deletion, deletion of non-existing buckets, and deletion failures due to dependencies or incorrect parameters.
5. Validate error responses for different failure scenarios.

### PR DESCRIPTION
**Pull request by kasabe28**
# Proposed Changes

- Add tests for `DeleteBucket` function Fixes #362

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,6 +63,7 @@ vet: ## Run go vet against code.
 .PHONY: test
 test: manifests generate fmt vet envtest checklicense ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go run github.com/onsi/ginkgo/v2/ginkgo --label-filter="!integration" -coverprofile cover.out ./...
+	go mod tidy
```
Comment (lukasfrank) [inline Makefile:66]: do we need this command?

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (kasabe28) [reply Makefile:66]: Yes. As it cleans test related dependencies from go.sum file after tests are run. This is added in other repositories as well like `gardener-extension-provider-onmetal`.

Comment (lukasfrank) [reply Makefile:66]: I think we should remove it since it seems to be a temporary fix. The pipeline also checks if the go.mod is up to-date.

Comment (kasabe28) [reply Makefile]: done

**Review by lukasfrank — APPROVED** [review]

**Pull request merged**